spec/wasabi/document/authentication_spec.rb

```diff
@@ -1,37 +1,46 @@
-require "spec_helper"
+require 'spec_helper'
 
 describe Wasabi::Document do
-  context "with: authentication.wsdl" do
-
-    subject { Wasabi::Document.new fixture(:authentication).read }
+  context 'with: authentication.wsdl' do
+    let(:document) { Wasabi::Document.new fixture(:authentication).read }
 
     describe '#namespace' do
-      subject { super().namespace }
-      it { should == "http://v1_0.ws.auth.order.example.com/" }
+      subject(:namespace) { document.namespace }
+
+      it { is_expected.to eq('http://v1_0.ws.auth.order.example.com/') }
     end
 
     describe '#endpoint' do
-      subject { super().endpoint }
-      it { should == URI("http://example.com/validation/1.0/AuthenticationService") }
+      subject(:endpoint) { document.endpoint }
+
+      it { is_expected.to eq URI('http%3A%2F%2Fexample.com%2Fvalidation%2F1.0%2FAuthenticationService') }
```

Comment:
This _seems_ like a change in behavior?

Reply:
@olleolleolle That's true, man...
Seems that the original idea is to transform something like this:
`http%3A%2F%2Fcartoons.com%2Fmgm%2Fold_cartoons.asmx%3Fcartoon%3DTom+%26+Jerry%26year%3D1986`
In this:
`http://cartoons.com/mgm/old_cartoons.asmx?cartoon=Tom%20&%20Jerry&year=1986`

Keeping escaped just query string after the URL...
```rb
url = 'http%3A%2F%2Fcartoons.com%2Fmgm%2Fold_cartoons.asmx%3Fcartoon%3DTom+%26+Jerry%26year%3D1986'
unescaped_url = CGI.unescape(url.to_s)
# => "http://cartoons.com/mgm/old_cartoons.asmx?cartoon=Tom & Jerry&year=1986"
parsed_url = URI(unescaped_url)
% => #<URI::HTTP http://cartoons.com/mgm/old_cartoons.asmx?cartoon=Tom%20&%20Jerry&year=1986>
```
Fixed [here](https://github.com/savonrb/wasabi/pull/75/commits/f6ab0a50900cde84f05c5ac09b82f48a11e54948).
Thanks.